MCR-2271 Abstract MCRMetadataStore structure

OPNA2608 opened the pull request:
[Link to jira](https://mycore.atlassian.net/browse/MCR-2271).

This PR has been in development for afew months and went through several different implementation ideas, I'm still in the middle of cleaning up leftover debugging code and unnecessary changes but I think it's ready for a general review by MyCoRe core developers. All tests pass locally and I could use an MCRMetadataStore-based application with this new code as-is, but I didn't do much testing on the SVN side of things.

---

This is a restructuring of the Metadata-Store <-> Metadata <-> Metadata-Version code.

- I created MCRMetadata, a storage-agnostic class to interface with Metadata-Stores.
This obsoletes the store-bound MCRStoredMetadata and MCRVersionedMetadata classes as a universal metadata object class. Most MCRMetadata operations map to abstract MCRMetadataStore ones that are left up to the associated store to implement. The term "revision" has been changed to mean an object's revision instead of a global SVN repository one.

- The Metadata-Store has been further abstracted.
MCRMetadataStore previously was an unversioned storage implementation using a local directory-file structure according to the specified slots. This new implementation has abstract CRUD methods that extending Metadata-Stores are supposed to implement, along with several convenience/compatibility methods that all wrap around to abstract ones. As far as this new MCRMetadataStore is concerned, all stores are assumed to handle versioning operations and information: a lack of versioning would be implemented by only ever listing & knowing one version of an object. The simple directory-file implementation has been extracted to MCRXMLMetadataStore & implements this.

- The SVN-based Metadata-Store has been renamed to MCRSVNXMLMetadataStore to better reflect its inheritance.
I tried to do a decoupled SVN-only store, but it would've been an even bigger undertaking to do this in a clean manner and didn't seem worth the effort. Along with the changed meaning of revision, this would've caused (more significant) performance drawbacks due to the SVNKit library lacking methods to conveniently get revisions of *files* as opposed to revisions of the *repository*.

- Some MCRMetadataVersion renaming/rewriting

- Changes throughout MyCoRe to accommodate the changed classes

---

I know this is a *big* PR, but I hope you're still able to give me some feedback on the ideas and changes proposed here. Cheers!

MyCoRe-Travis commented:
Something gone wrong on b6d0e03624d400701388e86d377300cf1410dd85! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6244-pull_request-1113

MyCoRe-Travis commented:
Something gone wrong on 1e7d9061287dbbd1a2544ff84127f9aeed987705! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6250-pull_request-1113

OPNA2608 committed:
MCR-2271 Abstract MCRMetadataStore structure

Begin to decouple MCRMetadataVersion from SVN.

OPNA2608 committed:
WIP MCR-2271 Abstract MCRMetadataStore structure

- WIP Merge MCRStoredMetadata & MCRVersionedMetadata into MCRMetadata class
- WIP Migrate MCRMetadataStore & MCRVersioningMetadataStore from
  storage-type-specific Metadata classes to united MCRMetadata

Plan for followup commits:
- MCRMetadataStore (MCRAbstractMetadataStore?): Abstract base class for metadata stores, abstract
  CRUD-methods
- XML-store -> MCRXMLMetadataStore
- SVN-store -> MCRSVNMetadataStore
- Move any store-specific code from MCRMetadata into stores, get
  MCRMetadata store-generic

MyCoRe-Travis commented:
Something gone wrong on cf8b51c835205f6f7dd30fd9bf13c93ddc11c527! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6488-pull_request-1113

OPNA2608 commented:
Essentially solved via the recent, smaller PRs.
